core/Archive/ArchiveInvalidator.php

```diff
@@ -528,20 +535,61 @@ public function removeInvalidations($idSite, $plugin, $report = null)
     }
 
     /**
-     * Re-archives reports without propagating exceptions.
+     * Schedules a re-archiving reports without propagating exceptions. This is scheduled
+     * since adding invalidations can take a long time and delay UI response times.
      *
      * @param int|int[]|'all' $idSites
      * @param string $pluginName
      * @param string|null $report
      * @param Date|null $startDate
      */
-    public function reArchiveReportSafely($idSites, string $pluginName, string $report = null, Date $startDate = null)
+    public function scheduleReArchiving($idSites, string $pluginName, string $report = null, Date $startDate = null)
     {
         try {
-            $this->reArchiveReport($idSites, $pluginName, $report, $startDate);
+            $reArchiveList = new ReArchiveList($this->logger);
+            $reArchiveList->add(json_encode([
+                'idSites' => $idSites,
+                'pluginName' => $pluginName,
+                'report' => $report,
+                'startDate' => $startDate ? $startDate->getTimestamp() : null,
+            ]));
         } catch (\Throwable $ex) {
-            $logger = StaticContainer::get(LoggerInterface::class);
-            $logger->info("Failed to schedule rearchiving of past reports for $pluginName plugin.");
+            $this->logger->info("Failed to schedule rearchiving of past reports for $pluginName plugin.");
+        }
+    }
+
+    /**
+     * Applies the queued archiving rearchiving entries.
+     */
+    public function applyScheduledReArchiving()
+    {
+        $reArchiveList = new ReArchiveList($this->logger);
+        $items = $reArchiveList->getAll();
+
+        foreach ($items as $entry) {
+            try {
+                $entry = @json_decode($entry, true);
+                if (empty($entry)) {
+                    continue;
+                }
+
+                $this->reArchiveReport(
+                    $entry['idSites'],
+                    $entry['pluginName'],
+                    $entry['report'],
+                    !empty($entry['startDate']) ? Date::factory($entry['startDate']) : null
```

Comment:
should we maybe cast `$entry['startDate']` to integer before passing it to the factory. It should normally already be an integer, but if it would be a string (containing a number), the factory will throw an exception

Reply:
👍